src/riscv/lib/src/pvm/tezos.rs

```diff
@@ -249,6 +267,58 @@ where
     Ok(hash.len() as u64)
 }
 
+/// Verify a Secp256k1 signature.
+#[inline]
+fn handle_tezos_secp256k1_verify<MC, M>(
+    machine: &mut MachineCoreState<MC, M>,
+) -> Result<u64, SbiError>
+where
+    MC: MemoryConfig,
+    M: ManagerReadWrite,
+{
+    let arg_pk_addr = machine.hart.xregisters.read(a0);
+    let arg_sig_addr = machine.hart.xregisters.read(a1);
+    let arg_msg_addr = machine.hart.xregisters.read(a2);
+
+    let pk_bytes: [u8; 65] = machine.main_memory.read(arg_pk_addr)?;
+    let sig_bytes: [u8; 64] = machine.main_memory.read(arg_sig_addr)?;
+    let msg_bytes: [u8; 32] = machine.main_memory.read(arg_msg_addr)?;
+
+    let pk = PublicKey::parse(&pk_bytes).map_err(|_| SbiError::Failed)?;
+    let sig = SecpSig::parse_standard(&sig_bytes).map_err(|_| SbiError::Failed)?;
+    let msg = Message::parse(&msg_bytes);
+    let valid = libsecp256k1::verify(&msg, &sig, &pk);
+
+    Ok(valid as u64)
+}
+
+/// Compute a Keccak 256-bit digest.
+#[inline]
+fn handle_tezos_keccak256_hash<MC, M>(
+    machine: &mut MachineCoreState<MC, M>,
+) -> Result<u64, SbiError>
+where
+    MC: MemoryConfig,
+    M: ManagerReadWrite,
+{
+    let arg_out_addr = machine.hart.xregisters.read(a0);
+    let arg_msg_addr = machine.hart.xregisters.read(a1);
+    let arg_msg_len = machine.hart.xregisters.read(a2);
+
+    if (arg_msg_len > MAX_PVM_MEMORY_ACCESS as u64) {
```

Comment:
Use the potentially larger integer to avoid truncation. Or alternatively, define `MAX_PVM_MEMORY_ACCESS` as a `u64`.

```suggestion
    if ((arg_msg_len as usize) > MAX_PVM_MEMORY_ACCESS) {
```